src/main/java/gregtech/api/pipenet/tile/TileEntityPipeBase.java

```diff
@@ -361,10 +364,8 @@ public void readFromNBT(@Nonnull NBTTagCompound compound) {
         if (compound.hasKey("InsulationColor")) {
             this.paintingColor = compound.getInteger("InsulationColor");
         }
-        String frameMaterialName = compound.getString("FrameMaterial");
-        if (!frameMaterialName.isEmpty()) {
-            this.frameMaterial = GregTechAPI.MATERIAL_REGISTRY.getObject(frameMaterialName);
-        }
+        this.frameMaterial = GregTechAPI.materialManager.getMaterial(compound.getString("FrameMaterial"));
```

Comment:
This should keep the empty check, because empty string being passed to this is one of the options